### ISSUE TITLE
Clean invalid data

### ISSUE THREAD

**Issue by Flaburgan** [jaywink/the-federation.info#90]
There are several invalid data in the current database I'd like to clean, to have more accurate numbers and better charts (especially because of the scale).

First thing, the pods without a network:
```
select host, version, network from pods where network = 'unknown';`
142 rows in set 
```
Most of them are diaspora* pods. Even if a lot of them are down, I still would like to clean the data. This would allow us to have the real number of pods, a correct sorting / icon displayed on the node list, and pods listed on the network page.

So I propose to run the following commands on the database:
```
UPDATE pods SET network = 'diaspora' WHERE network = 'unknown';
UPDATE pods SET network = 'friendica' WHERE host = 'friend.elsmussols.net' OR host = 'web.radiozagorje.eu' OR host = 'www.icirko.com' ;
```

(Other suggestions coming later)

**Comment by Flaburgan** [comment]
So, about the missing data, I spotted at least `2014-08-15`, 2016-09-26`, `2016-09-27` and `2016-10-30` which are missing for almost every nodes (`select * from stats where date = '2016-10-30'` returns only 28 entries). I guess something failed when pulling the data at these dates.

![capture du 2017-02-26 23-05-31](https://cloud.githubusercontent.com/assets/930064/23344194/4ce52fbe-fc78-11e6-815b-c8b00c834b14.png)

I propose to use the previous available data (so to put the value of 2016-10-30 at 2016-10-31 for the missing nodes) to avoid these peaks which destroy the chart, especially the scale of the calculate ones like posts per users:
![capture du 2017-02-26 23-09-23](https://cloud.githubusercontent.com/assets/930064/23344232/fbb4ee08-fc78-11e6-9be1-a2a3276eab4a.png)

@jaywink do you agree?

**Comment by Flaburgan** [comment]
Here is the queries to fix the peaks:
```
mysql> INSERT INTO stats(date, pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments)
    -> SELECT '2014-08-15', pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments FROM stats s2
    -> WHERE date = '2014-08-14' AND NOT EXISTS (SELECT 1 FROM stats s1 WHERE date = '2014-08-15' AND s1.pod_id = s2.pod_id);
Query OK, 127 rows affected (0,68 sec)
Records: 127  Duplicates: 0  Warnings: 0

mysql> INSERT INTO stats(date, pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments)
    -> SELECT '2016-09-26', pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments FROM stats s2
    -> WHERE date = '2016-09-25' AND NOT EXISTS (SELECT 1 FROM stats s1 WHERE date = '2016-09-26' AND s1.pod_id = s2.pod_id);
Query OK, 324 rows affected (0,86 sec)
Records: 324  Duplicates: 0  Warnings: 0

mysql> INSERT INTO stats(date, pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments)
    -> SELECT '2016-09-27', pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments FROM stats s2
    -> WHERE date = '2016-09-26' AND NOT EXISTS (SELECT 1 FROM stats s1 WHERE date = '2016-09-27' AND s1.pod_id = s2.pod_id);
Query OK, 319 rows affected (0,84 sec)
Records: 319  Duplicates: 0  Warnings: 0

mysql> INSERT INTO stats(date, pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments)
    -> SELECT '2016-10-30', pod_id, total_users, active_users_halfyear, active_users_monthly, local_posts, local_comments FROM stats s2
    -> WHERE date = '2016-10-29' AND NOT EXISTS (SELECT 1 FROM stats s1 WHERE date = '2016-10-30' AND s1.pod_id = s2.pod_id);
Query OK, 295 rows affected (0,78 sec)
Records: 295  Duplicates: 0  Warnings: 0
```

And the charts are now looking like that:
![capture du 2017-02-27 13-01-57](https://cloud.githubusercontent.com/assets/930064/23360677/f2d8d9d8-fcec-11e6-8b41-705e81683050.png)
![capture du 2017-02-27 13-01-39](https://cloud.githubusercontent.com/assets/930064/23360679/f5d85f96-fcec-11e6-94e0-476bc4a0f2b2.png)

Better, isn't it?

**Comment by jaywink** [comment]
> Most of them are diaspora* pods. Even if a lot of them are down, I still would like to clean the data. 

Out of the 141 rows only 11 are actually up and running. They're all diaspora pods. So sure, we can fix the data, but the code will just change it back on next update (for these 11) - here: https://github.com/jaywink/the-federation.info/blob/c4378f5ff11d05f66391811697ce91ed6f6690f4/src/database.js#L41-L41

TBH, I think assuming everything is a diaspora pod for the future is a bit dangerous, since it's only legacy data that fits this. If another server stack suddenly implements the protocol but it doesn't supply a network key, then we're making bad assumptions.

I'd say those 11 should upgrade their software instead of the-federation.info having to maintain extra code to figure out their software stack.

As for the missing stats, yeah, the polling mechanism needs a little rewrite, it's very unstable at the moment and sometimes just bonkers out, causing data to be missed or duplicated (upwards peaks). Need to fix the mechanism separately, I'll run your queries to fix the worst peaks, thanks! 👍

**Comment by Flaburgan** [comment]
>I think assuming everything is a diaspora pod for the future is a bit dangerous

I don't want that at all, I just wanted to do that at the time we run the query. But I didn't know that the pod table was updated daily, I thought only the stats one was updated.

**Comment by jaywink** [comment]
Closed via #95
